pennylane_lightning_gpu/src/simulator/StateVectorCudaManaged.hpp

```diff
@@ -601,6 +607,20 @@ class StateVectorCudaManaged
         applyDeviceMatrixGate(mat.data(), {}, wires, adjoint);
     }
 
+    /* Multi-qubit gates */
+    inline void applyMultiRZ(const std::vector<std::size_t> &wires,
+                             bool adjoint, Precision param) {
+        static const std::vector<std::string> names(wires.size(), {"RZ"});
+        applyParametricPauliGate(names, {}, wires, param, adjoint);
+    }
+    inline void applyGeneratorMultiRZ(const std::vector<std::size_t> &wires,
+                                      bool adjoint) {
+        static const std::string name{"PauliZ"};
+        static const Precision param = 0.0;
+        applyDeviceMatrixGate(gate_cache_.get_gate_device_ptr(name, param), {},
```

Comment:
Doesn't `applyDeviceMatrixGate` need a matrix with size $2^{\rm nTargets} \times 2^{\rm nTargets}$ as in https://docs.nvidia.com/cuda/cuquantum/custatevec/api/functions.html#custatevecapplymatrix ?

Reply:
Thank you @chaeyeunpark! You're right. It shouldn't be this; it's been fixed while testing but I forgot to commit it here.